### PR TITLE
Pass only TrackLocation instead of whole vehicle

### PR DESCRIPTION
**Pull request by hdpoliveira**

### CODE REVIEW

File: src/openrct2/ride/Vehicle.cpp [hunk 1]
```diff
@@ -7989,7 +7988,7 @@ bool Vehicle::UpdateTrackMotionForwardsGetNewTrack(uint16_t trackType, Ride* cur
                 }
             }
             map_invalidate_element(TrackLocation, tileElement);
-            vehicle_update_block_brakes_open_previous_section(this, tileElement);
+            block_brakes_open_previous_section(*curRide, TrackLocation, tileElement);
```
Comment (tupaschoal) [inline src/openrct2/ride/Vehicle.cpp:7991]: Are we certain that `curRide` is valid in here?

Comment (Gymnasiast) [reply src/openrct2/ride/Vehicle.cpp:7991]: This is also not exactly how it worked previously. It used `vehicle->ride`, not `curRide`.

Comment (hdpoliveira) [reply src/openrct2/ride/Vehicle.cpp:7991]: > Are we certain that `curRide` is valid in here?

@tupaschoal Yes, it is verified some lines above.

> This is also not exactly how it worked previously. It used `vehicle->ride`, not `curRide`.

@Gymnasiast Yes, the actual Ride object was being recalculated within the function without need.

Comment (tupaschoal) [reply src/openrct2/ride/Vehicle.cpp:7991]: Ok then. @Gymnasiast you cool with this?

Comment (Gymnasiast) [reply src/openrct2/ride/Vehicle.cpp:7991]: I want to check it myself.

Comment (Gymnasiast) [reply src/openrct2/ride/Vehicle.cpp:7991]: OK, I traced it back. `curRide` is snowballed from `Vehicle::UpdateTrackMotion`, which exits if it is null. It is indeed set from `vehicle->ride`.

**Pull request merged**